src/preact.d.ts

```diff
@@ -11,7 +11,8 @@ declare namespace preact {
 	interface PreactHTMLAttributes {
 		dangerouslySetInnerHTML?:DangerouslySetInnerHTML;
 		key?:string;
-		ref?:(el?: Element) => void;
+		defaultValue?:string;
+		ref?:<T,S>(component:Component<T,S>) => void;
```

Comment:
fixed in https://github.com/developit/preact/pull/350/commits/791cfd4c0195c5cce872fa235da8386054b0db7e